onnxruntime/core/providers/openvino/openvino_graph.cc

```diff
@@ -0,0 +1,369 @@
+// Copyright(C) 2019 Intel Corporation
+// Licensed under the MIT License
+
+#include <iostream>
+#include <cstdlib>
+#include <map>
+#include <string>
+#include <memory>
+#include <cstdlib>
+#include <fstream>
+#include <Python.h>
+
+#include <inference_engine.hpp>
+#include <ie_builders.hpp>
+
+#include "core/graph/graph.h"
+#include "core/framework/tensorprotoutils.h"
+
+#include "openvino_graph.h"
+
+namespace openvino_ep {
+
+OpenVINOGraph::OpenVINOGraph(onnxruntime::Node* fused_node, std::string /*device_info*/) {
+	//TODO: parse device info to obtain the following values
+
+
+  device_id_ = "CPU";
+  precision_ = InferenceEngine::Precision::FP32;
+  std::string precision_str = "FP32";
+
+#ifdef OPENVINO_CONFIG_CPU_FP32
+	device_id_ = "CPU";
+	precision_ = InferenceEngine::Precision::FP32;
+	precision_str = "FP32";
+#endif
+#ifdef OPENVINO_CONFIG_GPU_FP32
+	device_id_ = "GPU";
+	precision_ = InferenceEngine::Precision::FP32;
+	precision_str = "FP32";
+#endif
+#ifdef OPENVINO_CONFIG_GPU_FP16
+	device_id_ = "GPU";
+	precision_ = InferenceEngine::Precision::FP16;
+	precision_str = "FP16";
+#endif
+#ifdef OPENVINO_CONFIG_MYRIAD
+	device_id_ = "MYRIAD";
+	precision_ = InferenceEngine::Precision::FP16;
+	precision_str = "FP16";
+#endif
+#ifdef OPENVINO_CONFIG_VAD_R
+	device_id_ = "HDDL";
+	precision_ = InferenceEngine::Precision::FP16;
+	precision_str = "FP16";
+#endif
+
+  std::cout<< "[OpenVINO-EP] device:" << device_id_ << std::endl;
+  std::cout<< "[OpenVINO-EP] precision:" << precision_str << std::endl;
+
+
+	num_inf_reqs_ = (device_id_ == "HDDL") ? 8 : 1;
+
+	fused_node_ = fused_node;
+
+
+
+    cnn_network_ = BuildCNNNetworkWithMO();
+
+  // TODO: make this a debug option
+  // Uncomment the below code to enable dumping IE CNN graphs.
+  //std::string file_name = "./conv_" + fused_node->Name();
+  //cnn_network_->serialize( file_name+".xml", file_name+".bin");
+
+
+  infer_requests_ = GetExecutableHandle(cnn_network_, device_id_, precision_);
+}
+
+std::vector<std::string> OpenVINOGraph::GetEnvLdLibraryPath() {
+    std::string plugin_path = std::getenv("LD_LIBRARY_PATH");
+    std::vector<std::string> paths;
+    std::string token;
+    std::istringstream tokenStream(plugin_path);
+    char delimiter = ':';
+
+    while (std::getline(tokenStream , token, delimiter)) {
+      paths.push_back(token);
+    }
+    return paths;
+}
+
+std::shared_ptr<InferenceEngine::CNNNetwork> OpenVINOGraph::BuildCNNNetworkWithMO() {
```

Comment:
are we only supporting CNN based architectures?

Reply:
Though OpenVINO was originally designed for CV based graphs, it now supports non-CNN type architectures too. The 'CNNNetwork' mentioned here is just the name of an API data structure left unchanged from the older design.

Reply:
This will cause confusion. Any suggestions?

Reply:
We can certainly rename the local function 'BuildCNNNetworkWithMO' and the names of the local handles to the data structure instances.
The type name 'InferenceEngine::CNNNetwork', however, is from the OpenVINO's API. We might want to try to not hide it completely as it exposes certain attributes of the translated graph that we consume ahead in the code. We could typedef it to an another name, that'll add an indirection layer. Do you think this could reduce the confusion?

Reply:
All associated local identifiers now use the term 'OpenVINO Network' to avoid this confusion. The InferenceEngine::CNNNetwork have been left as is as these are the names from OpenVINO's public API. Would it be okay to keep it this way?